src/compiler/scala/tools/nsc/ast/parser/Scanners.scala

```diff
@@ -498,8 +498,11 @@ trait Scanners extends ScannersCommon {
             nextChar()
             ch match {
               case 'x' | 'X' => base = 16 ; nextChar()
-              case _         => base = 8    // single decimal zero, perhaps
+              case 'b' | 'B' => base = 2  ; nextChar()
```

Comment:
I'm a little bit concerned about making this change before the SIP is accepted

Reply:
I'll spit it out, I mean split it out. BTW, I think it would work better as SPIT, the Scala Process to Improve Things.